onnxruntime/test/testdata/transform/cast_elimination.py

```diff
@@ -0,0 +1,48 @@
+import onnx
+from onnx import helper
+from onnx import TensorProto, GraphProto, OperatorSetIdProto
+from onnx import numpy_helper
+import numpy as np
+
+X1 = helper.make_tensor_value_info('x1', TensorProto.INT64, [4, 4])
+X2 = helper.make_tensor_value_info('x2', TensorProto.INT64, [4, 1])
+X3 = helper.make_tensor_value_info('x3', TensorProto.INT64, [4, 1])
+Y = helper.make_tensor_value_info('output', TensorProto.INT64, [4, 4])
+
+less1 = helper.make_node('Less', ['x1', 'x2'], ['less1'], name='less1')
+less2 = helper.make_node('Less', ['x1', 'x3'], ['less2'], name='less2')
+cast1 = helper.make_node('Cast', ['less1'], ['cast1'], name='cast1', to=9)
+and_node = helper.make_node('And', ['cast1', 'less2'], ['and_node'], name='and_node')
+cast2 = helper.make_node('Cast', ['and_node'], ['cast2'], name='cast2', to=9)
+cast3 = helper.make_node('Cast', ['cast2'], ['cast3'], name='cast3', to=1)
+cast4 = helper.make_node('Cast', ['x1'], ['cast4'], name='cast4', to=1)
+matmul = helper.make_node('MatMul', ['cast3', 'cast4'], ['matmul'], name='matmul')
+cast5 = helper.make_node('Cast', ['matmul'], ['cast5'], name='cast5', to=7)
+cast6 = helper.make_node('Cast', ['cast5'], ['cast6'], name='cast6', to=7)
+identity = helper.make_node('Identity', ['cast6'], ['output'], name='identity')
```

Comment:
cast as graph outputs is also an important test case.

Reply:
Added some more Cast nodes around input/output for test.